src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/MigrationController.kt

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.digital.hmpps.authorizationserver.resource
+
+import com.microsoft.applicationinsights.TelemetryClient
+import jakarta.validation.Valid
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.ResponseStatus
+import uk.gov.justice.digital.hmpps.authorizationserver.config.AuthenticationFacade
+import uk.gov.justice.digital.hmpps.authorizationserver.config.trackEvent
+import uk.gov.justice.digital.hmpps.authorizationserver.service.MigrationClientService
+import java.time.Instant
+import java.time.LocalDate
+
+@Controller
+class MigrationController(
+  private val migrationClientService: MigrationClientService,
+  private val telemetryClient: TelemetryClient,
+  private val authenticationFacade: AuthenticationFacade,
+) {
+
+  @PostMapping("migrate-client")
+  @ResponseStatus(HttpStatus.OK)
+  @PreAuthorize("hasRole('ROLE_OAUTH_ADMIN')")
+  fun addClient(
+    @Valid @RequestBody
+    clientDetails: MigrationClientRequest,
+  ): ResponseEntity<Any> {
+    val registrationResponse = migrationClientService.addClient(clientDetails)
```

Comment:
Question - do we need to capture the response here? The service does not return anything (think default return type in Kotlin is Unit?). Would it be clearer to define the end point as not returning anything?

Reply:
removed response, it was required at all as it already have annotation  @ResponseStatus(HttpStatus.OK)